jerry-core/mem/mem-allocator.cpp

```diff
@@ -53,26 +53,7 @@ mem_finalize (bool is_show_mem_stats) /**< show heap memory stats
 
   if (is_show_mem_stats)
   {
-    mem_heap_print (false, false, true);
-
-#ifdef MEM_STATS
-    mem_pools_stats_t stats;
-    mem_pools_get_stats (&stats);
-
-    printf ("Pools stats:\n");
-    printf (" Chunk size: %zu\n"
-            "  Pools: %zu\n"
-            "  Allocated chunks: %zu\n"
-            "  Free chunks: %zu\n"
-            "  Peak pools: %zu\n"
-            "  Peak allocated chunks: %zu\n\n",
-            MEM_POOL_CHUNK_SIZE,
-            stats.pools_count,
-            stats.allocated_chunks,
-            stats.free_chunks,
-            stats.peak_pools_count,
-            stats.peak_allocated_chunks);
-#endif /* MEM_STATS */
+    mem_stats_print ();
```

Comment:
@akiss77, I see. Thank you for your explanation.
The part of `mem_heap_print` that is not currently guarded by `MEM_STATS`, is not executed during the call, because the corresponding arguments are set to `false`, and only `dump_stats` argument is set to `true`.
So, maybe, its better to extract part of `mem_heap_print` that outputs statistics to another routine like `mem_heap_print_stats`, guard it with `MEM_STATS`, and call it from `mem_heap_print` and `mem_stats_print`. In the case, `mem_stats_print` could be guarded by `#ifdef` around the function, not inside.
What do you think about this?